js/foundation.offcanvas.js

```diff
@@ -87,8 +89,14 @@ class OffCanvas {
       'keydown.zf.offcanvas': this._handleKeyboard.bind(this)
     });
 
-    if (this.options.closeOnClick && this.$exiter.length) {
-      this.$exiter.on({'click.zf.offcanvas': this.close.bind(this)});
+    // If we have an overlay and close on click, let it close the off canvas menu.
+    if (this.options.closeOnClick && this.options.contentOverlay && this.$overlay.length) {
```

Comment:
Why are we using truthy/falsy here as compared to direct true/false comparison like on line 98?